### PR TITLE
Delete CoreFX tests of COM lifetime control methods.

### LINKED ISSUE #38252: Test failure:System.Runtime.InteropServices.Tests.AreComObjectsAvailableForCleanupTests/AreComObjectsAvailableForCleanup_NothingToCleanup_ReturnsFalse
Test `System.Runtime.InteropServices.Tests.AreComObjectsAvailableForCleanupTests/AreComObjectsAvailableForCleanup_NothingToCleanup_ReturnsFalse ` has failed.

Message :
```
Assert.False() Failure
Expected: False
Actual:   True
```

Stack Trace :
```
   at System.Runtime.InteropServices.Tests.AreComObjectsAvailableForCleanupTests.AreComObjectsAvailableForCleanup_NothingToCleanup_ReturnsFalse() in /_/src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/AreComObjectsAvailableForCleanupTests.cs:line 14
```

Build: -[20190604.76](https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fheads~2Fmaster/test~2Ffunctional~2Fcli~2Finnerloop~2F/20190604.76)(Master)

Failing configurations:

- Windows.10.Arm64.Open-arm64-Release

Details:
https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fheads~2Fmaster/test~2Ffunctional~2Fcli~2Finnerloop~2F/20190604.76/workItem/System.Runtime.InteropServices.Tests/analysis/xunit/System.Runtime.InteropServices.Tests.AreComObjectsAvailableForCleanupTests~2FAreComObjectsAvailableForCleanup_NothingToCleanup_ReturnsFalse

### PR DESCRIPTION
**Pull request by jkoritzinsky**
These tests are not always stable on multi-threaded test runs and can cause spurious failures. https://github.com/dotnet/coreclr/pull/24990 adds tests of these methods in CoreCLR, where we can have stable test runs of their functionality.

Fixes #38252

### CODE REVIEW

**Comment by stephentoub** [comment]
> adds tests of these methods in CoreCLR, where we can have stable test runs of their functionality

Why can't they be stable here? You can disable concurrent test execution here if that's the problem.

**Comment by AaronRobinsonMSFT** [comment]
>You can disable concurrent test execution here if that's the problem.

I didn't know that was possible. How is that done?

I think in general these kind of COM lifetime related tests are best handled in the coreclr repo since the vast majority of COM semantics are handled at the runtime level. Definitely not against having tests here as well though.

**Comment by stephentoub** [comment]
> How is that done?

By adding an attribute like this:
https://github.com/dotnet/corefx/blob/master/src/System.Linq.Parallel/tests/XunitAssemblyAttributes.cs

**Comment by jkoritzinsky** [comment]
The thing I don't like about that is that we'd have to make the whole assembly run sequentially since there's not a good way to say "tests that create COM objects can't run in parallel with these tests". There's also the fact that some of these tests (specifically the `Thread` test and the `AreComObjectAvailableForCleanup` test) weren't really testing the functionality.

**Comment by stephentoub** [comment]
> The thing I don't like about that is that we'd have to make the whole assembly run sequentially since there's not a good way to say "tests that create COM objects can't run in parallel with these tests".

There is. https://github.com/xunit/xunit/issues/318

**Comment by jkoritzinsky** [comment]
The hard problem is tracking down all of the tests that create COM objects and making sure we always mark them. It would be tedious and error-prone when we add new tests to ensure that we always mark any test that could create a COM object.

**Comment by stephentoub** [comment]
> The hard problem is tracking down all of the tests that create COM objects and making sure we always mark them.

I'm not sure why.  I would hope they'd only be on a few test classes, and those classes could be restricted in this way: any tests added to those classes in the future would inherit it.  You seem to be suggesting we should never have a COM object in a test anywhere in corefx, which sounds quite limiting.

If you believe that fundamentally such tests should be in coreclr instead of corefx because of how what being tested relates to what functionality the runtime provides and whether those are implementation details or not, that's fine, they can be in coreclr.  But please don't move the tests because of perceived infrastructure differences.

**Comment by jkoritzinsky** [comment]
I feel that these specific tests are testing features related to the functionality the runtime provides and expose implementation details of the runtime and how it handles RCW lifetime. @AaronRobinsonMSFT can chime in on whether or not he feels the same. I also feel that the tests that are currently in CoreFX aren't actually getting us much test coverage of these methods since they're basically only testing degenerate no-op cases.

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/AreComObjectsAvailableForCleanupTests.cs [hunk 1]
```diff
@@ -1,17 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
```
Comment (stephentoub) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/AreComObjectsAvailableForCleanupTests.cs]: The references to these files would also need to be removed from the .csproj.

**Comment by AaronRobinsonMSFT** [comment]
>You seem to be suggesting we should never have a COM object in a test anywhere in corefx, which sounds quite limiting.

@stephentoub I don't think anyone is suggesting that. Sorry if that is how the message was interpreted. COM tests in CoreFX are more than likely fine in most scenarios. The one area where I think testing in CoreCLR is around COM object lifetime _or_ to be more specific when reliance or direct interaction with CCWs and RCWs is being validated. Debugging and simply guaranteeing specific behavior is easy closer to the runtime than CoreFX provides.

>But please don't move the tests because of perceived infrastructure differences.

I agree with this entirely. I think the reasoning here is specifically that it is easy to test and validate with higher certainty these lifetime scenarios from the CoreCLR repo.

**Comment by stephentoub** [comment]
Ok. Thanks for the explanation.

**Pull request merged**